doc/jit.md

```diff
@@ -0,0 +1,44 @@
+# Testarossa JIT (tr.jit) compiler
+
+Testarossa JIT compiler is a high performant dynamic compiler that powers openj9 - an enterprise-grade Java Virtual Machine (JVM). It's characteristic features are to compress the application code for improved runtime performance by exploiting the exact host CPU model and apply profiling on the runtime behavior of the generated code, feed it back to the optimizer to further refine it. The component provides a wide range of command line options to (i) trace the actions of the compilation and the compiled code, and (ii) control the compiler behavior.
+
+While care is taken at the design itself to derive optimal performance under default configurations for the most common workloads, these command line options help to derive insight into the efficiency of the compiler for a given workload, and fine tune the compilation behavior for maximizing the runtime performance / throughput.
+
+
+This page contains the full list of compiler options that are classified into: (i) optimization control knob options (which generally takes the form of enable | disable syntax), (ii) trace options (which generally take the form of traceXXX syntax), and (iii) debug options (which generally helps to manually control the execution on a specific compilation / compiled code event) and (iv) Environment variables that influences the behavior of JVMs that are spawned within that environment.
+
+For most part of the documentation, this below code will be used.
+
+```java
+public class Foo {
+  public static void main(String args[]) {
+    new Foo().run(Integer.parseInt(args[0]));
+  }
+  public void run(int count) {
+    for(int i=0; i<count; i++)
+      System.out.println("hello jit world!");
+  }
+}
+```
+
+## Control options
+
+### Disable JIT
```

Comment:
I almost think it would be best to document `-Xtrace` first :)

Reply:
thanks - sure, I will try to place a one liner for `Xtrace`. I will keep it minimal as:
 - `-Xtrace` usage in this document will be limited,
 - the limited usage of it here will be mostly intuitive to the readers.